src/main/java/io/cdap/plugin/http/source/common/pagination/BaseHttpPaginationIterator.java

```diff
@@ -185,7 +174,11 @@ public PageEntry next() {
       throw new NoSuchElementException("No more pages to load.");
     }
 
-    return new PageEntry(httpStatusCode, page.next());
+    try {
+      return new PageSuccessEntry(httpStatusCode, page.next(), page.getStringVersionOfRecord());
```

Comment:
a page entry should not require a string representation of a record. It seems like this is being done in order to pass something to the transform method.

This iterator should not return PageEntry objects, but Page objects. The transform method should get Page objects from the InputFormat and call a method that will iterate through the records contained in the Page, emitting errors if needed. The PageEntry abstraction can be removed.

Reply:
> a page entry should not require a string representation of a record. 
I use these text entries which failed to transform to structured record as a part of "send to error" routine. They are basically the body of InvalidEntry. 
This is useful so user can let's say handle his invalid json entries, which failed due to types or missing field.

Reply:
Done.